app/assets/javascripts/project.js

```diff
@@ -1,38 +1,24 @@
-    $("#one_year_project, #six_months_project, #three_months_monthly_project, #six_months_monthly_project, #one_year_monthly_project").hide();
-    $('input[name="options_project"]').change( function() {
-      ProjectDashboard.update_chart($(this).val(), $('input[name="radio_project"]').filter(':checked').val());
+    $('input[name="project_filter"]').attr("autocomplete", "off");
+    $('.project').hide().filter('#three_months_project').show();
+    //This is used to change the chart based on button 3 months, 6months and 1 year.
+    $('input[name="project_options"]').change( function() {
+      ProjectDashboard.update_chart($('input[name="project_filter"]').filter(':checked').val());
     })
-    $('input[name="radio_project"]').change( function() {
-      ProjectDashboard.update_chart($('input[name="options_project"]').filter(':checked').val(), $(this).val());
+    //This is used to filter the chart based on weekly and monthly.
+    $('input[name="project_filter"]').change( function() {
+      ProjectDashboard.update_chart($(this).val());
     })
   },
-  update_chart: function(months, filter) {
-    if(months == '3' && filter == 'weekly') {
-      $("#three_months_project").show();
-      $("#one_year_project, #six_months_project, #three_months_monthly_project, #six_months_monthly_project, #one_year_monthly_project").hide();
+  update_chart: function(filter) {
+    if(filter == 'monthly') {
+      var contClass = $('input[name="project_options"]').filter(':checked').data('div');
+      $('.project').hide().filter('#' + contClass + '_monthly_project').show();
     }
-    else if(months == '6' && filter == 'weekly'){
-      $("#six_months_project").show();
-      $("#one_year_project, #three_months_project, #three_months_monthly_project, #six_months_monthly_project, #one_year_monthly_project").hide();
-    }
-    else if(months == '12' && filter == 'weekly') {
-      $("#one_year_project").show();
-      $("#six_months_project, #three_months_project, #three_months_monthly_project, #six_months_monthly_project, #one_year_monthly_project").hide();
-    }
-    else if(months == '3' && filter == 'monthly') {
-      $("#three_months_monthly_project").show();
-      $("#six_months_project, #three_months_project, #one_year_project, #six_months_monthly_project, #one_year_monthly_project").hide();
-    }
-    else if(months == '6' && filter == 'monthly') {
-      $("#six_months_monthly_project").show();
-      $("#three_months_project, #six_months_project, #one_year_project, #three_months_monthly_project, #one_year_monthly_project").hide();
-    }
-    else if(months == '12' && filter == 'monthly') {
-      $("#one_year_monthly_project").show();
-      $("#three_months_project, #six_months_project, #one_year_project, #three_months_monthly_project, #six_months_monthly_project").hide();
+    else if(filter == 'weekly') {
+      var contClass = $('input[name="project_options"]').filter(':checked').data('div');
```

Comment:
This line can be called once and passed into the function.  Otherwise looks good

Reply:
I tried calling that the line at once, But problem is button checked value is changing dynamically and when we click on button change function is getting called.

So, both the change function I need to call that line again.